### PR TITLE
Fix some bug

### PR DESCRIPTION
**Pull request by iattempt**
(Put an `X` inside the `[ ]` to denote check mark `[X]`.)

- [ ] **If creating a new file :**
  - [ ] added links to it in the README files ?
  - [ ] included tests with it ?
  - [ ] added description (overview of algorithm, time and space compleixty, and possible edge case) in docstrings ?

- [x] **if done some changes :**
  - [x] wrote short description in the PR explaining what the changes do ?
  - [ ] Fixes #[issue number] if related to any issue

- [ ] **other**

1. Fix bugs: compare None with `is` or `is not` not boolean.
2. Fix coding style: Remove some redundant spaces and lines.
3. add restrict conditions for test code if it has been write in source file.

### CODE REVIEW

File: algorithms/tree/is_balanced.py [hunk 1]
```diff
@@ -1,33 +1,32 @@
...
-    left = max_height(root.left)
-    right = max_height(root.right)
-    return abs(left-right) <= 1 and is_balanced(root.left) and is_balanced(root.right)
+# def is_balanced(root):
+#     """
+#     O(N^2) solution
+#     """
+#     left = max_height(root.left)
+#     right = max_height(root.right)
+#     return abs(left-right) <= 1 and is_balanced(root.left) and is_balanced(root.right)
```
[the remark this answers is not in the document]
Comment (iattempt) [reply algorithms/tree/is_balanced.py]: @goswami-rahul it is N^2.

File: algorithms/tree/bin_tree_to_list.py [hunk 2]
```diff
@@ -32,18 +30,21 @@ def bin_tree_to_list_util(root):
...
-tree.right = Node(15)
-tree.left.left  = Node(25)
-tree.left.left.right  = Node(100)
-tree.left.right = Node(30)
-tree.right.left = Node(36)
 
-head = bin_tree_to_list(tree)
-print_tree(head)
+if __name__ == '__main__':
+    tree = TreeNode(10)
```
[the remark this answers is not in the document]
Comment (keon) [reply algorithms/tree/bin_tree_to_list.py]: sure, I didn't get to refactor trees yet, but yeah we should remove it.

File: algorithms/tree/is_balanced.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (keon) [reply algorithms/tree/is_balanced.py]: I think it'd be better to be renamed `is_balanced_recursive`

[the remark this answers is not in the document]
Comment (iattempt) [reply algorithms/tree/is_balanced.py]: The recursive one is private now, and it will easy to be changed if iterative one is added.

**Pull request merged**